database/crdt-tweet-repo.go

```diff
@@ -0,0 +1,170 @@
+		}
+	}
+
+	return retweeted, nil
+}
+
+// UnRetweet removes a retweet and updates CRDT counter
+func (repo *CRDTTweetRepo) UnRetweet(retweetedByUserID, tweetId string) error {
+	// Remove retweet locally first
+	err := repo.TweetRepo.UnRetweet(retweetedByUserID, tweetId)
+	if err != nil {
+		return err
+	}
+
+	// Update CRDT counter
+	if repo.crdtStore != nil {
+		_, err := repo.crdtStore.DecrementStat(tweetId, crdt.StatTypeRetweets)
+		if err != nil {
+			log.Warnf("failed to decrement CRDT retweet count: %v", err)
+		}
+	}
+
+	return nil
+}
+
+// RetweetsCount returns the aggregated retweet count from CRDT if available
+func (repo *CRDTTweetRepo) RetweetsCount(tweetId string) (uint64, error) {
+	if tweetId == "" {
+		return 0, local.DBError("empty tweet id")
+	}
+
+	// Try to get aggregated count from CRDT first
+	if repo.crdtStore != nil {
+		count, err := repo.crdtStore.GetAggregatedStat(tweetId, crdt.StatTypeRetweets)
+		if err != nil {
+			log.Warnf("failed to get CRDT retweet count, falling back to local: %v", err)
+			return repo.TweetRepo.RetweetsCount(tweetId)
+		}
+		return count, nil
+	}
+
+	// Fallback to local count
+	return repo.TweetRepo.RetweetsCount(tweetId)
+}
+
+// Create adds a new tweet and increments view counter
+func (repo *CRDTTweetRepo) Create(userId string, tweet domain.Tweet) (domain.Tweet, error) {
+	created, err := repo.TweetRepo.Create(userId, tweet)
+	if err != nil {
+		return domain.Tweet{}, err
+	}
+
+	// Initialize view counter in CRDT (optional)
+	if repo.crdtStore != nil {
+		// Views will be tracked when tweets are fetched
+	}
+
+	return created, nil
+}
+
+// CreateWithTTL adds a tweet with TTL and initializes CRDT counters
+func (repo *CRDTTweetRepo) CreateWithTTL(userId string, tweet domain.Tweet, duration time.Duration) (domain.Tweet, error) {
+	created, err := repo.TweetRepo.CreateWithTTL(userId, tweet, duration)
+	if err != nil {
+		return domain.Tweet{}, err
+	}
+
+	return created, nil
+}
+
+// IncrementViewCount increments the view counter for a tweet using CRDT
+func (repo *CRDTTweetRepo) IncrementViewCount(tweetId string) (uint64, error) {
+	if tweetId == "" {
+		return 0, local.DBError("empty tweet id")
+	}
+
+	if repo.crdtStore != nil {
+		count, err := repo.crdtStore.IncrementStat(tweetId, crdt.StatTypeViews)
+		if err != nil {
+			log.Warnf("failed to increment CRDT view count: %v", err)
+			return 0, err
+		}
+		return count, nil
+	}
+
+	return 0, local.DBError("CRDT store not available")
+}
+
+// GetViewCount returns the aggregated view count from CRDT
+func (repo *CRDTTweetRepo) GetViewCount(tweetId string) (uint64, error) {
+	if tweetId == "" {
+		return 0, local.DBError("empty tweet id")
+	}
+
+	if repo.crdtStore != nil {
+		count, err := repo.crdtStore.GetAggregatedStat(tweetId, crdt.StatTypeViews)
+		if err != nil {
+			log.Warnf("failed to get CRDT view count: %v", err)
+			return 0, err
+		}
+		return count, nil
+	}
+
+	return 0, local.DBError("CRDT store not available")
+}
```

Comment:
You left code of "tweet-repo" unrewised. Now it contains duplicated functionality. Refactor it according to main plan.